pkg/rpcclient/nns/contract.go

```diff
@@ -31,6 +36,28 @@ type ContractReader struct {
 	hash    util.Uint160
 }
 
+// Actor is used by Contract to call state-changing methods.
+type Actor interface {
+	Invoker
+
+	nep11.Actor
+
+	MakeCall(contract util.Uint160, method string, params ...any) (*transaction.Transaction, error)
+	MakeRun(script []byte) (*transaction.Transaction, error)
+	MakeUnsignedCall(contract util.Uint160, method string, attrs []transaction.Attribute, params ...any) (*transaction.Transaction, error)
+	MakeUnsignedRun(script []byte, attrs []transaction.Attribute) (*transaction.Transaction, error)
+	SendCall(contract util.Uint160, method string, params ...any) (util.Uint256, uint32, error)
+	SendRun(script []byte) (util.Uint256, uint32, error)
+}
+
+// Contract implements all contract methods.
+type Contract struct {
```

Comment:
Have you compared existing methods with mainnet NNS? Our goal is to have all methods compatible with it. If needed, then existing methods may be changed.

Reply:
yes, i hope i didn't miss any of them